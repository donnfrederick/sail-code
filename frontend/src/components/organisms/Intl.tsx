import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
