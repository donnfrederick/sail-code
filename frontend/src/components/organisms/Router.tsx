import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
