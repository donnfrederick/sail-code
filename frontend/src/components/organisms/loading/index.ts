import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
