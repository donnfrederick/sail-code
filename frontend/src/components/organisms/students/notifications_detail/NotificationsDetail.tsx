import Button from 'components/atoms/students/Button'
import UserProfile from 'components/organisms/students/user_profile'
import withNotificationsId from 'hocs/withNotificationsId'
import * as ConversationModels from 'models/conversation'
import * as NotificationModels from 'models/notification'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'

interface Props {
  cancelledConversation: ConversationModels.CancelledConversation | null
  conversation: ConversationModels.Conversation | null
  myId: number
  notification: NotificationModels.Notification
}

export default withNotificationsId((props: Props) => {
  const { cancelledConversation, conversation, myId, notification } = props

  const cancelledConversationExists =
    cancelledConversation && Object.values(cancelledConversation).length

  const isCancelledConversation =
    !conversation && cancelledConversationExists ? true : false
  const relatedConversation = isCancelledConversation
    ? cancelledConversation
    : conversation
  const partner = relatedConversation
    ? getPartner(relatedConversation, myId)
    : null

  const notificationText = NotificationModels.notificationText.student
  const additionalNotification =
    notificationText[notification.notification_type]

  const Buttons = () => {
    const CancelReservationButton = () =>
      conversation ? (
        <Button
          type="blue"
          text="Details"
          width={300}
          height={112}
          fontSize={36}
          link={resolvePath.page(
            'students',
            `conversations/detail/${conversation.id}`
          )}
        />
      ) : null

    const NewReservationButton = () => (
      <Button
        type="blue"
        text="Reserve"
        width={300}
        height={112}
        fontSize={36}
        link={resolvePath.page('students', 'reservations/new')}
      />
    )

    const ActionButton = () =>
      notification.notification_type ===
      NotificationModels.NotificationTypes.cancellation ? (
        <NewReservationButton />
      ) : (
        <CancelReservationButton />
      )

    return (
      <ButtonContainer>
        <ActionButton />
      </ButtonContainer>
    )
  }

  return (
    <Container>
      <NotificationContent>
        <TimeStamp>
          {moment.parseZone(notification.notificated_at).format('MMMM D, YYYY')}
        </TimeStamp>
        <Title>{notification.title}</Title>
        <Text>{notification.body}</Text>
      </NotificationContent>
      {partner ? <UserProfile user={partner} type="basic" /> : null}
      <AdditionalInfo>
        {additionalNotification ? (
          <AdditionalText>{additionalNotification}</AdditionalText>
        ) : null}
        {isCancelledConversation ? (
          <CancellationText>
            {notificationText.cancelledConversation}
          </CancellationText>
        ) : null}
      </AdditionalInfo>
      <Buttons />
    </Container>
  )
})

const Container = styled.div`
  width: 100%;
  margin: 0 auto;
`

const NotificationContent = styled.div`
  text-align: left;
  margin-bottom: 50px;
`

const TimeStamp = styled.time`
  display: block;
  margin-bottom: 32px;
  font-size: 28px;
  font-weight: 500;
  line-height: 1;
  letter-spacing: 0.5px;
  color: #90aec1;
`

const Title = styled.h2`
  margin-bottom: 36px;
  font-size: 40px;
  font-weight: 500;
  line-height: 1.35;
  letter-spacing: 0.5px;
  color: #405766;
`

const Text = styled.p`
  margin: 0;
  padding: 0;
  font-size: 32px;
  font-weight: 500;
  line-height: 1.5;
  white-space: pre-wrap;
  color: #405766;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin: 0 auto;
  padding-bottom: 56px;
`

const AdditionalInfo = styled.div`
  margin: 40px 0 72px;
`

const AdditionalText = styled(Text)`
  margin-bottom: 72px;
`

const CancellationText = styled(Text)`
  color: red;
`
