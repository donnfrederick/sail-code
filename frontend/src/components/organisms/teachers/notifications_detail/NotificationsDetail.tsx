import Button from 'components/atoms/teachers/Button'
import UserProfile from 'components/organisms/teachers/user_profile'
import withNotificationsId from 'hocs/withNotificationsId'
import * as ConversationModels from 'models/conversation'
import * as NotificationModels from 'models/notification'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'

interface Props {
  notification: NotificationModels.Notification
  cancelledConversation: ConversationModels.CancelledConversation | null
  conversation: ConversationModels.Conversation | null
  myId: number
}

export default withNotificationsId((props: Props) => {
  const { notification, cancelledConversation, conversation, myId } = props

  const cancelledConversationExists =
    cancelledConversation && Object.values(cancelledConversation).length

  const isCancelledConversation =
    !conversation && cancelledConversationExists ? true : false
  const relatedConversation = isCancelledConversation
    ? cancelledConversation
    : conversation

  const partner = relatedConversation
    ? getPartner(relatedConversation, myId)
    : null

  const notificationText = NotificationModels.notificationText.teacher
  const additionalNotification =
    notificationText[notification.notification_type]

  const Buttons = () => {
    const CancelReservationButton = () =>
      conversation ? (
        <Button
          type="blue"
          text="詳細"
          width={300}
          height={112}
          fontSize={36}
          link={resolvePath.page(
            'teachers',
            `reservations/detail/${conversation.id}`
          )}
        />
      ) : null

    const NewReservationButton = () => (
      <Button
        type="blue"
        text="予約する"
        width={300}
        height={112}
        fontSize={36}
        link={resolvePath.page('teachers', 'reservations/new')}
      />
    )

    const SecondaryButton = () =>
      notification.notification_type ===
      NotificationModels.NotificationTypes.cancellation ? (
        <NewReservationButton />
      ) : (
        <CancelReservationButton />
      )

    return (
      <ButtonContainer>
        <Button
          type="white"
          text="戻る"
          width={300}
          height={112}
          fontSize={36}
          link={resolvePath.page('teachers', 'notifications')}
        />
        <SecondaryButton />
      </ButtonContainer>
    )
  }

  return (
    <>
      <NotificationContent>
        <TimeStamp>
          {moment
            .parseZone(notification.notificated_at)
            .format('YYYY/MM/DD HH:mm')}
        </TimeStamp>
        <Title>{notification.title}</Title>
        <Text>{notification.body}</Text>
      </NotificationContent>
      {partner ? <UserProfile user={partner} type="basic" /> : null}
      <AdditionalInfo>
        {additionalNotification ? <Text>{additionalNotification}</Text> : null}
        {isCancelledConversation ? (
          <CancellationText>
            {notificationText.cancelledConversation}
          </CancellationText>
        ) : null}
      </AdditionalInfo>
      <Buttons />
    </>
  )
})

const NotificationContent = styled.div`
  width: 640px;
  margin: 0 auto;
  text-align: left;
`

const TimeStamp = styled.time`
  display: block;
  margin-bottom: 48px;
  font-size: 32px;
  font-weight: 500;
  letter-spacing: 0px;
  color: #90aec1;
`

const Title = styled.h2`
  margin: 0 0 48px;
  font-size: 40px;
  font-weight: 500;
  line-height: 1.4;
  letter-spacing: 0px;
  color: #405766;
`

const Text = styled.p`
  margin: 0 0 72px;
  padding: 0;
  font-size: 36px;
  font-weight: 500;
  line-height: 1.61;
  letter-spacing: 0px;
  white-space: pre-wrap;
  color: #405766;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin: 0 auto;
  padding-bottom: 56px;
`

const AdditionalInfo = styled.div`
  margin: 40px 40px 72px;
`

const CancellationText = styled(Text)`
  color: red;
`
