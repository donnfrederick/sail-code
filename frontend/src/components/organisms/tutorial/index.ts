import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
