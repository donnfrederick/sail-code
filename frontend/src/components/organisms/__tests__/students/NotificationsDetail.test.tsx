import Router from 'components/organisms/Router'
import NotificationsDetail from 'components/organisms/students/notifications_detail'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <NotificationsDetail />', () => {
  const tree = renderer
    .create(
      <Intl>
        <Router>
          <NotificationsDetail />
        </Router>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
