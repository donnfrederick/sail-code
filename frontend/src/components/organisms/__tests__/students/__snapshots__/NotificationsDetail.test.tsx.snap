// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <NotificationsDetail /> 1`] = `
<div
  className="sc-gGBfsJ kmQxcA"
>
  <div
    className="sc-jnlKLf iVXAMz"
  >
    <time
      className="sc-fYxtnH kJamYY"
    >
      Invalid date
    </time>
    <h2
      className="sc-tilXH kAzzxe"
    >
      
    </h2>
    <p
      className="sc-hEsumM izZQFJ"
    >
      
    </p>
  </div>
  <div
    className="sc-cIShpX ePKtiJ"
  />
  <div
    className="sc-ktHwxA kJpZCy"
  />
</div>
`;
